packages/devtools_app/lib/src/screens/inspector_v2/inspector_screen_body.dart

```diff
@@ -94,10 +94,10 @@ class InspectorScreenBodyState extends State<InspectorScreenBody>
         searchPreventClose = false;
       }
     });
-    addAutoDisposeListener(preferences.inspector.pubRootDirectories, () {
+    addAutoDisposeListener(preferences.inspector.pubRootDirectories, () async {
       if (serviceConnection.serviceManager.connectedState.value.connected &&
           controller.firstInspectorTreeLoadCompleted) {
-        controller.refreshInspector();
+        await controller.refreshInspector();
```

Comment:
I believe it is better to await the tree refresh here to avoid race conditions

Reply:
Is the closure awaited when the listener is called in the auto dispose logic? If not, this await won't matter and we should add a comment or explicitly mark unawaited

Reply:
Ah good point, it's not. Switched to `unawaited`